rfcs/react-components/convergence/semantic-shared-colors.md

```diff
@@ -0,0 +1,195 @@
+- 👍 Partners can override persona and status colors independently.
+- 👎 We will remove some "basic" colors `green`, `yellow` - are we sure those have been used as semantic colors only?
+  approach
+- 👎 1A: adds ~30 new tokens
+- 👎 If design decides to do the same for **presence** tokens, we will need to add another ~40 tokens to follow the same
+- 👎 1A: can cause temporarily inconsistent UI colors between library and partners
+- 👍 1B: we can use the current mapping in v1 theme and the new mapping in v2 theme - partners can control the moment when they switch to the new mapping
+- 👎 1B: requires us maintaining the double amount of themes
+- 👎 1C: breaks partners (the questions is how much, but...)
+
+### 2. Continue with the current approach
+
+The current names are set for v9, let's live with them 🤷‍♂️.
+
+Add "status" slots to `cranberry`, add `orange` as a status color.
+
+Update the sources in the library to use the new colors. Ask partners to update their sources.
+
+#### Pros and Cons
+
+- 👍 Minimal change, no new patterns, just adds 15 new tokens.
+- 👎 Can cause temporarily inconsistent UI colors between library and partners.
+- 👎 Not improving designers' workflow.
+
+### 3. Hack the hex values
+
+The current names are set for v9, let's live with them 🤷‍♂️.
+
+Change the hex values for `red` to `cranberry` values, `yellow` to `orange`.
+
+#### Pros and Cons
+
+- 👍 Change does not affect components or partner code.
+- 👍 No new tokens.
+- 👎 Designers will scream as "Yellow and orange are two different colors!".
+- 👎 There will be `red` and `cranberry` persona colors in the theme, but they will have the same hex values (but sure, we can take `cranberry` and change its hex values to, let's say `red` 🤯).
+
+### 4. Add semantic color tokens mapped to the current shared colors
+
+All tokens are mapped to CSS variables with the same name: `colorPaletteRedBackground1` token is mapped to `--colorPaletteRedBackground1` CSS variable. From the very first release of FUIR9, we [always warned partners](https://react.fluentui.dev/?path=/docs/concepts-developer-theming--page#do-not-use-css-variables-directly) that they should never depend on this fact:
+
+> Do not use CSS variables directly
+>
+> ⚠ Never use theme CSS variables directly! The CSS variables implementation of the theme is internal to the library. We might eventually decide to change the variable names, hash them or even use direct values instead of some variables. Always use the tokens to access the theme.
+
+Add "status" slots to `cranberry`, add `orange` as a status color.
+
+Add semantic color tokens which are mapped to the current shared colors. For example, `colorPaletteDangerBackground1` is mapped to `--colorPaletteRedBackground1`. `Danger` is still mapped to the old `red` color, but the token name is now semantic. Still not changed to `cranberry`.
+
+Update the sources in the library to use the new colors.
+
+Update the mapping from old colors to the new colors.
+
```

Comment:
````suggestion
```ts
const tokens = {
  // new tokens (note: naming is not final)
  colorStatusDangerBackground1: 'var(--colorPalleteRedBackground1)',
  colorStatusWarningBackground1: 'var(--colorPalletedYellowBackground1)', 
  
  colorPalleteRedBackground1: 'var(--colorPalleteRedBackground1'),
  colorPaletteYellowBackground1: 'var(--colorPaletteYellowBackground1'),
}

const webLightTheme = {
  colorPaletteGreenBackground1: 'HEX_VALUE',
  colorPaletteYellowBackground1: 'HEX_VALUE',
}
```
````